### PR TITLE
feat(Dialog): All variants use `flex`

### LINKED ISSUE #1268: Dialog should use `flex` on all variants

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Changes

- `.iui-dialog` has `display: flex;` & `flex-direction: column;` rather than the class modifiers.
- Removed `max-height` from `.iui-dialog-content`, added a `max-height` using `dvh` to `.iui-dialog-default` in response to https://github.com/iTwin/iTwinUI/pull/1269#issuecomment-1542670933.

Closes #1268.

## Testing

- Ran visual tests.
- Checked in to ensure overflow is working as expected in browser & iOS Simulator.

## Docs

N/A

### CODE REVIEW

File: .changeset/thirty-lemons-tie.md
Comment (mayank99) [inline .changeset/thirty-lemons-tie.md:5]:
```suggestion
All dialog variants have `flex` applied by default. This means the content should be wrapped with `Dialog.Content` or `ModalContent` for optimal layout.
```

File: .changeset/thirty-lemons-tie.md
Comment (FlyersPh9) [reply .changeset/thirty-lemons-tie.md:2]: Ok, suggestion applied.

File: .changeset/thirty-lemons-tie.md
Comment (FlyersPh9) [reply .changeset/thirty-lemons-tie.md]: Ok, suggestion applied.

**Review by mayank99 — CHANGES_REQUESTED** [review]
LGTM. Lets wait for pending v2 PRs to release (#1250) before merging this.

(Blocking accidental merge using "Request changes".)

**Comment by mayank99** [comment]
@FlyersPh9 I vaguely recall there were some hacks we needed to do because `display: flex` was not always present. Think you'll be able to dig through the history and find those?

Here's one I found: https://github.com/iTwin/iTwinUI/pull/1066#issuecomment-1424330775

We might want to make some changes in the dialog code now that we can assume `display: flex` is always present.

**Comment by FlyersPh9** [comment]
> @FlyersPh9 I vaguely recall there were some hacks we needed to do because `display: flex` was not always present. Think you'll be able to dig through the history and find those?
> 
> Here's one I found: [#1066 (comment)](https://github.com/iTwin/iTwinUI/pull/1066#issuecomment-1424330775)
> 
> We might want to make some changes in the dialog code now that we can assume `display: flex` is always present.

Done, added a bullet to the PR description with more deets.

**Pull request merged**
